lara-typescript/src/interactive-api-client/client.ts

```diff
@@ -132,6 +132,14 @@ export class Client {
     return this.removeListener("customMessage");
   }
 
+  public addDecorateContentListener(callback: ICustomMessageHandler) {
```

Comment:
I wonder why this is using `ICustomMessageHandler`.  Since this is listening for its own event it seems it should have its own callback type.

Reply:
Yeah, I think that is wrong.

Reply:
I'm going to have to study this a little more.  I'm quite confused by the types.

Reply:
Should it just be an iframePhone.ListenerCallback?  Or should it use something that fits the message?
```
export interface ITextDecorationInfo {
  words: string[];
  replace: string;
  wordClass: string;
  eventListeners: IEventListeners;
}
```

Reply:
addListener inside the client wants an argument of type iframePhone.ListenerCallback for the second argument.  So maybe I'm not understanding how to set up a listener with a different type.

Reply:
Ah, so I assume you mean the type that is passed into the callback itself.

Reply:
Ok, I think I sorted it out.  The last commit should have added types.

Reply:
Sorry I don't know the details either. It just struck me as wrong that an `ICustomMessageHander` was being passed to a `addDecoratedContentListener`.   customMessage is a special iframe message name/type that is different than the a decorateContent or decoratedContentEvent iframe messages.

Reply:
Check out the last commit.  I have to jump on a meeting, but I think it's sorted out.